test/gleam/float_test.gleam

```diff
@@ -375,28 +375,52 @@ pub fn random_test() {
     float.random(10.0, -10.0)
     |> fn(x) { x >=. -10.0 && x <. 10.0 }
     |> should.be_true
+
+    // two consecutive double precision floats
+    float.random(1.0, 1.0000000000000002)
+    |> fn(x) { x >=. 1.0 && x <. 1.0000000000000002 }
+    |> should.be_true
+
+    float.random(1.0000000000000002 , 1.0)
+    |> fn(x) { x >. 1.0 && x <=. 1.0000000000000002 }
+    |> should.be_true
+
+    // two consecutive double precision floats to test possible rounding error
+    // 9007199254740993 rounds to 9007199254740992 when in float
+    // 9007199254740995 rounds to 9007199254740996 when in float
+    float.random(9007199254740992.0, 9007199254740994.0)
+    |> should.equal(9007199254740992.0)
+
+    float.random(9007199254740994.0, 9007199254740992.0)
+    |> should.equal(9007199254740994.0)
+
+    float.random(9007199254740994.0, 9007199254740996.0)
+    |> should.equal(9007199254740994.0)
+
+    float.random(9007199254740996.0, 9007199254740994.0)
+    |> should.equal(9007199254740996.0)
```

Comment:
These new tests will check for the rounding issue mentioned. Going with @inoas suggestion to ensure correctness, we do extra checks in the `random` for floats and `int` (javascript)